src/crypto/algorithms/base.js

```diff
@@ -140,6 +140,26 @@ class DecryptionAlgorithm {
     importRoomKey(session) {
         // ignore by default
     }
+
+    /**
+     * Determine if we have the keys necessary to respond to a room key request
+     *
+     * @param {module:crypto#RoomKeyRequest} keyRequest
+     * @return {boolean} true if we have the keys and could (theoretically) share
```

Comment:
If it's in local storage, won't we hit size limits again? It's pretty easy to convert this to be `q(bool)` right? I would prefer if we did an async design up-front, as it's often a pain to turn a synchronous API into an async one retrospectively.